crates/host/src/module.rs

```diff
@@ -266,109 +231,99 @@ impl SerializedModuleCache {
                 // instantiated with fresh stores free from state. Instance
                 // creation is highly performant which makes caching of instances
                 // and stores unnecessary.
-                let module = unsafe {
+                //
+                // See https://github.com/wasmerio/wasmer/issues/4377
```

Comment:
@jost-s can you confirm that this is the relevant issue?

Reply:
No, it's this discussion https://github.com/wasmerio/wasmer/discussions/3829#discussioncomment-5790763
```suggestion
                // See https://github.com/wasmerio/wasmer/discussions/3829#discussioncomment-5790763
```